### PR TITLE
Add README to explain how to add new settings

### PR DESCRIPTION
**Pull request by shabalind**

### CODE REVIEW

File: BenchmarksCore/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+## Adding new settings 
+
+Let's walk through the process of adding a new String-typed setting:
+
+1. Add a new setting to BenchmarkSettings.swift:
```
Comment (texasmichelle) [inline BenchmarksCore/README.md:5]:
```suggestion
1. Add a new setting to [BenchmarkSettings.swift](https://github.com/tensorflow/swift-models/blob/master/BenchmarksCore/BenchmarkSettings.swift):
```

File: BenchmarksCore/README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      var mySetting: String {
+        if let value = self[MySetting.self]?.value {
+          return value
+        } else {
+          fatalError("MySetting setting must have a default.")
+        }
+      }
+    ```
+
+4. Add a new benchmark flag to BenchmarkArguments.swift:
```
Comment (texasmichelle) [inline BenchmarksCore/README.md:49]:
```suggestion
4. Add a new benchmark flag to [BenchmarkArguments.swift](https://github.com/tensorflow/swift-models/blob/master/BenchmarksCore/BenchmarkArguments.swift):
```

File: BenchmarksCore/README.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    @Flag(help: "Useful description of MySetting here.")
+    var mySettingFlagName: String?
+    ```
+
+    b. Validate your flag value in `BenchmarkArguments.validate()` function if necessary. 
+
+    c. Convert command-line flag to a setting in `BenchmarkArgument.settings` computed property:
+
+    ```swift
+      if value = mySettingFlagName {
```
Comment (texasmichelle) [inline BenchmarksCore/README.md:63]:
```suggestion
      if let value = mySettingFlagName {
```

File: BenchmarksCore/README.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        }
+      }
+    ```
+
+4. Add a new benchmark flag to BenchmarkArguments.swift:
+
+    a. Add a flag property:
+
+    ```swift
+    @Flag(help: "Useful description of MySetting here.")
```
Comment (texasmichelle) [inline BenchmarksCore/README.md:54]:
```suggestion
    @Argument(help: "Useful description of MySetting here.")
```

File: BenchmarksCore/README.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        }
+      }
+    ```
+
+4. Add a new benchmark flag to [BenchmarkArguments.swift](https://github.com/tensorflow/swift-models/blob/master/BenchmarksCore/BenchmarkArguments.swift):
+
+    a. Add a flag property:
+
+    ```swift
+    @Option(help: "Useful description of MySetting here.")
```
Comment (texasmichelle) [inline BenchmarksCore/README.md:54]:
```suggestion
    @Option(name: .customLong("mySettingFlagName"), help: "Useful description of MySetting here.")
```

Comment (shabalind) [reply BenchmarksCore/README.md:54]: swift-benchmark prefers default non-camelCase names, while swift-models have been using camelCase previously. can/shall standardise on moving away from camelCase going forward? /cc @BradLarson

Comment (texasmichelle) [reply BenchmarksCore/README.md:54]: Yeah, I guess there's no point having this in the docs if we want to remove it in #636 👍

Comment (shabalind) [reply BenchmarksCore/README.md:54]: Sgtm!

**Comment by shabalind** [comment]
@texasmichelle Now that #642 is merged, I dropped the recommendation to use `.customLong`.

**Pull request merged**